### PR TITLE
integration test for the jira issue https://wso2.org/jira/browse/ESBJ…

### PR DESCRIPTION
**Pull request by lankavitharana**
…AVA-5030

### CODE REVIEW

File: integration/mediation-tests/tests-patches/src/test/resources/testng.xml [hunk 1]
```diff
@@ -255,6 +255,7 @@
     <test name="Payload Factory" preserve-order="true" verbose="2" parallel="false">
         <classes>
             <class name="org.wso2.carbon.esb.mediators.payloadFactory.ESBJAVA3573PayloadFormatWithBeginHtmlTagArgument"/>
+            <!--<class name="org.wso2.carbon.esb.mediators.payloadFactory.ESBJAVA5030PayloadFormatArgumentWithTrailingSpaceTestCase"/>--> <!-- TODO uncomment once dependencies changes to SNAPSHOT-->
```
Comment (vinok88) [inline integration/mediation-tests/tests-patches/src/test/resources/testng.xml]: Why is the test case commented? :)

Comment (milindaperera) [inline integration/mediation-tests/tests-patches/src/test/resources/testng.xml]: Why this test case is commented?

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/mediators/payloadFactory/ESBJAVA5030PayloadFormatArgumentWithTrailingSpaceTestCase.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                         "  \"input\": { \"value\": \"<abc>sample</abc> \" }\n" +
+                         "}";
+
+        Reader data = new StringReader(payload);
+        Writer writer = new StringWriter();
+
+        String serviceURL = this.getApiInvocationURL("trailingSpaceAPI");
+
+        String response = HttpURLConnectionClient.sendPostRequestAndReadResponse(data,
+                                                                                 new URL(serviceURL), writer, "application/json");
```
Comment (milindaperera) [inline integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/mediators/payloadFactory/ESBJAVA5030PayloadFormatArgumentWithTrailingSpaceTestCase.java]: Formatting

File: integration/mediation-tests/tests-patches/src/test/resources/testng.xml (hunk 1 above)
Comment (lankavitharana) [inline integration/mediation-tests/tests-patches/src/test/resources/testng.xml]: EI is not getting build on top of synapse snapshot versions, hence public branch fix doesn't come to the EI snapshot build, hence it's commented, later when it is migrated to snapshot dependency we can uncomment this, (TODO is for that)

**Pull request merged**